code/__HELPERS/icons.dm

```diff
@@ -981,9 +981,10 @@ GLOBAL_LIST_EMPTY(friendly_animal_types)
 	var/static/list/humanoid_icon_cache = list()
 	if(icon_id && humanoid_icon_cache[icon_id])
 		return humanoid_icon_cache[icon_id]
-
-	var/mob/living/carbon/human/dummy/body = generate_or_wait_for_human_dummy(dummy_key)
-
+// SKYRAT EDIT BEGIN - Icon Rendering Corruption Fix
+	var/mob/living/mob_to_copy = get_mob_by_key(dummy_key)
```

Comment:
Won't this always return null?

Reply:
nope, somehow it works. This is what the detailed examine and the character editor do to draw a character.

Reply:
This proc returns a mob whose `ckey` is equal to whatever is passed to it. In the case of the photo records, the thing passed to it is the string `"dummy_manifest_generation"`. Which means this thing is returning null, unless someone is logged in with the username of `"dummy_manifest_generation"`

Reply:
The original used this "dummy_key" successfully to get the client - and this also successfully grabs it aswell. The way this proc is called seems to overwrite it. I didn't change anything about how the proc is called, and am simply reusing dummy key. It works somehow.

Reply:
The issue with the original line is that while it rendered the first mob in the round correctly, every one generated afterwards would have the parts of the prior on them - and - and the next - and the next - and the next. Eventually, it would just make blank icons. With this change, that doesn't happen - you get the correct image for each mob.

Reply:
/proc/generate_dummy_lookalike returns a blank dummy if you pass it `null, null`. In addition the first argument should be a string not a mob.

Reply:
Actually, your change is pretty much identical to what it would be if you went to `get_id_photo` and made it pass null to the proc, since /proc/generate_dummy_lookalike just calls the same `generate_or_wait_for_human_dummy` that was being called before with the first argument if you give it null as the second argument. In other words your change is pretty much creating a new dummy every time, bypassing the problem rather than actually fixing it. Also it's not deleting the dummy afterwards, since the dummy_key is not null, it won't delete it at the end now.

Reply:
¯\_(ツ)_/¯ It still works somehow.

Reply:
This is the proc that pretty much every instance of character preview generation uses now. 

Ever since https://github.com/Skyrat-SS13/Skyrat-tg/pull/4933 this has been the proper proc to generate a new icon. Only code last touched 4 years ago or so uses the older proc.

Reply:
generate_or_wait_for_human_dummy doesn't even take two arguments

Reply:
Right, in earlier commits it *only* returned random humans, so this _has_ to be doing something correct. And before the earlier commits, it returned a mangled mess of everyone's parts before the current.

Reply:
What your thing is "doing correct" is completely avoiding the problem instead of fixing it by just creating a brand new human dummy every time instead of reusing it. Which is okay, but the reason it's creating a brand new human dummy every time is not because that's how you intended to fix it but because you used the proc incorrectly.

Here's `generate_dummy_lookalike`:
```dm
/proc/generate_dummy_lookalike(slotkey, mob/target)
	if(!istype(target))
		return generate_or_wait_for_human_dummy(slotkey)

	// the rest of the code
```

As you can see, if target is null it just calls generate_or_wait_for_human_dummy anyways. In addition, the first argument is not a mob.

Reply:
even if the passed argument isn't a mob, it still generated the correct icon for each mob with multiple players connected. 

and the part after that istype portion still runs, I believe. Otherwise, the outcome would be the same as the OG code?

Reply:
Aight, I smoke tested it. Doesn't even use the arguments like you said. Works as intended without them, but makes a new dummy every time from scratch. It probably cleans up, smoke testing the cleanup lines worked.